aws_lambda_powertools/utilities/feature_flags/time_conditions.py

```diff
@@ -0,0 +1,56 @@
+from datetime import datetime, tzinfo
+from typing import Dict, Optional
+
+from dateutil.tz import gettz
+
+from .schema import HOUR_MIN_SEPARATOR, TimeValues
+
+
+def _get_now_from_timezone(timezone: Optional[tzinfo]) -> datetime:
+    """
+    Returns now in the specified timezone. Defaults to UTC if not present.
+    At this stage, we already validated that the passed timezone string is valid, so we assume that
+    gettz() will return a tzinfo object.
+    """
+    timezone = gettz("UTC") if timezone is None else timezone
+    return datetime.now(timezone)
+
+
+def compare_days_of_week(action: str, values: Dict) -> bool:
+    timezone_name = values.get(TimeValues.TIMEZONE.value, "UTC")
+
+    # %A = Weekday as locale’s full name.
+    current_day = _get_now_from_timezone(gettz(timezone_name)).strftime("%A").upper()
+
+    days = values.get(TimeValues.DAYS.value, [])
+    return current_day in days
+
+
+def compare_datetime_range(action: str, values: Dict) -> bool:
+    timezone_name = values.get(TimeValues.TIMEZONE.value, "UTC")
+    timezone = gettz(timezone_name)
+    current_time: datetime = _get_now_from_timezone(timezone)
+
+    start_date_str = values.get(TimeValues.START.value, "")
+    end_date_str = values.get(TimeValues.END.value, "")
+
+    # Since start_date and end_date doesn't include timezone information, we mark the timestamp
+    # with the same timezone as the current_time. This way all the 3 timestamps will be on
+    # the same timezone.
+    start_date = datetime.fromisoformat(start_date_str).replace(tzinfo=timezone)
+    end_date = datetime.fromisoformat(end_date_str).replace(tzinfo=timezone)
+    return start_date <= current_time <= end_date
+
+
+def compare_time_range(action: str, values: Dict) -> bool:
+    timezone_name = values.get(TimeValues.TIMEZONE.value, "UTC")
+    current_time: datetime = _get_now_from_timezone(gettz(timezone_name))
+
+    start_hour, start_min = values.get(TimeValues.START.value, "").split(HOUR_MIN_SEPARATOR)
+    end_hour, end_min = values.get(TimeValues.END.value, "").split(HOUR_MIN_SEPARATOR)
+    return (
+        current_time.hour >= int(start_hour)
+        and current_time.hour <= int(end_hour)
+        and current_time.minute >= int(start_min)
+        and current_time.minute <= int(end_min)
+    )
```

Comment:
This code can cause problems when using start and end times with exact times, for example, start: 10:00 am and end: 11:00 am. This is also valid for any start and/or end where the minutes are 00. See the code example I made below:

![image](https://user-images.githubusercontent.com/4295173/212570054-103cf54c-fdb9-43b3-b189-05be033add38.png)

I fixed this code locally and it is working as expected. My only concern here is why do I need to remove datetime type for the current_time variable because `strftime` returns a string. If we want to keep typing, we need to convert the dates using `strptime`, but from what I see you have removed that function due to performance.

```suggestion
    timezone_name = values.get(TimeValues.TIMEZONE.value, "UTC")
    current_hour = _get_now_from_timezone(gettz(timezone_name)).strftime("%H:%M")

    start_hour = values.get(TimeValues.START.value, "")
    end_hour = values.get(TimeValues.END.value, "")

    return start_hour <= current_hour <= end_hour
```

Reply:
Thanks for this @leandrodamascena ! I digg a little further and doing a lexical comparison would also fail. However, I've addressed the problem by doing a little bit more work. I've updated the code and added some explanation. I also added a few new tests to test your scenario and two new ones. Please let me know how it looks now :)